.github/workflows/clang-format.yml

```
@@ -0,0 +1,54 @@
+name: Clang Format Check
+on:
+  pull_request:
+    paths:
+      - 'Sources/**'
+      - 'Tests/**'
+      - 'Samples/Common/Sources/CrashTriggers/**'
+      - '.github/workflows/clang-format.yml'
+      - '.clang-format-ignore'
+
+jobs:
+  formatting-check:
+    name: Formatting Check
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v4
+
+    - name: Install clang-format-18
+      run: |
+        wget https://apt.llvm.org/llvm.sh
+        chmod +x llvm.sh
+        sudo ./llvm.sh 18
+        sudo apt-get install -y clang-format-18
+
+    - name: Define directories
+      run: echo "DIRS='Sources Tests Samples/Common/Sources/CrashTriggers' " >> $GITHUB_ENV
+
+    - name: Check formatting
+      run: |
+        find $DIRS -name '*.c' -o -name '*.cpp' -o -name '*.h' -o -name '*.m' | \
+        xargs -r clang-format-18 -style=file -n -Werror
+
+    - name: Generate formatting diff
+      if: failure()
+      run: |
+        find $DIRS -name '*.c' -o -name '*.cpp' -o -name '*.h' -o -name '*.m' | \
```

I know we don't have ObjC++ in the lib so far, but there is such file in the sample. And also in case they are added to the lib it's better to have them covered here.
```suggestion
        find $DIRS -name '*.c' -o -name '*.cpp' -o -name '*.h' -o -name '*.m' -o -name '*.mm' | \
        xargs -r clang-format-18 -style=file -n -Werror

    - name: Generate formatting diff
      if: failure()
      run: |
        find $DIRS -name '*.c' -o -name '*.cpp' -o -name '*.h' -o -name '*.m' -o -name '*.mm' | \
```